yapapi/services/service.py

```diff
@@ -365,20 +361,19 @@ async def reset(self) -> None:
         Handlers of a restarted service are called more then once - all of the cleanup necessary between calls
         should be implemented here. E.g. if we initialize a counter in :func:`Service.__init__` and increment it
         in :func:`Service.start()`, we might want to reset it here to the initial value.
-
-        Target implementation (0.10.0 and up) will raise NotImplementedError. Current implementation only warns about
-        this future change.
         """
+        pass
 
-        msg = (
-            "Default implementation of Service.reset will raise NotImplementedError starting from 0.10.0. "
-            f"You should implement this method on {type(self)}"
+    @property
+    def restart_condition(self) -> bool:
+        """Dictated condition based on which this instance will be restarted."""
+        return (
+            self.exc_info != (None, None, None) and not self.__service_instance.started_successfully
         )
-        warn_deprecated_msg(msg)
 
     @property
     def is_available(self):
-        """Return `True` iff this instance is available (that is, starting, running or stopping)."""
+        """Return `True` if this instance is available (that is, starting, running or stopping)."""
```

Comment:
I believe the doubled `f` was intentional ... but yeah, the difference is miniscule indeed